### PR TITLE
Removed the regression test package [Final part]

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### Why is this the best possible solution? Were any other approaches considered?
This is the final part of https://github.com/getodk/collect/pull/5983 which contains commits 30-42.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It doesn't require testing.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] added any new strings with date formatting to `DateFormatsTest`
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formmanagement/BlankFormListTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+@RunWith(AndroidJUnit4::class)
+class BlankFormListTest {
+    private val rule = CollectTestRule()
+
+    @get:Rule
+    val chain: RuleChain = TestRuleChain.chain().around(rule)
+
+    @Test
+    fun sortByDialog_ShouldBeTranslatedAndDisplayProperIcons() {
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/formmanagement/BlankFormListTest.kt]: I think we can ditch this test. We use string resources for the options (and Android Lint would shout at us if we didn't). I don't think this is worth dedicating a connected test to.

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/CatchFormCalculationExceptionsTest.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@RunWith(AndroidJUnit4::class)
+class CatchFormCalculationExceptionsTest {
+
+    private val rule = CollectTestRule()
+
+    @get:Rule
+    val ruleChain: RuleChain = TestRuleChain.chain().around(rule)
+
+    @Test
+    fun typeMismatchErrorMessage_shouldBeDisplayedWhenItOccurs() {
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/CatchFormCalculationExceptionsTest.kt]: Could this go in `CatchFormDesignExceptionsTest` instead?

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormMediaTest.kt [hunk 3]
```diff
@@ -5,26 +5,61 @@ import org.junit.Rule
...
             .assertText("ab")
             .assertText("ac")
     }
+
+    @Test
+    fun missingFileMessage_shouldBeDisplayedIfExternalFileIsMissing() {
+        val formsDirPath = StoragePathProvider().getOdkDirPath(StorageSubdirectory.FORMS)
+
+        rule.startAtMainMenu()
+            .copyForm("search_and_select.xml")
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormMediaTest.kt]: I think we could split these out into their relevant select tests (`ExternalSelectsTest`, `ExternalSecondaryInstanceTest` etc)

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormCalculationsTest.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+@RunWith(AndroidJUnit4::class)
+class FormCalculationsTest {
+    private val rule = CollectTestRule()
+
+    @get:Rule
+    val ruleChain: RuleChain = TestRuleChain.chain().around(rule)
+
+    @Test
+    fun randomFunction_ShouldWorkCorrectly() {
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormCalculationsTest.kt]: This feels like it should be covered in JavaRosa? Does Collect actually implement anything to do with `random()`?

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormAppearancesTest.kt [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+@RunWith(AndroidJUnit4::class)
+class FormAppearancesTest {
+    private val rule = CollectTestRule()
+
+    @get:Rule
+    val ruleChain: RuleChain = TestRuleChain.chain().around(rule)
+
+    @Test
+    fun searchAppearance_ShouldDisplayWhenSearchAppearanceIsSpecified() {
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormAppearancesTest.kt]: I get that this is trying to make sure `search` works everywhere it should work (which makes sense to have in one test), but can we cut it down in any way? I don't think we need to assert on every result at this level right?

**Review by seadowg — CHANGES_REQUESTED** [review]

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formmanagement/BlankFormListTest.kt (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/androidTest/java/org/odk/collect/android/feature/formmanagement/BlankFormListTest.kt]: Done.

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/CatchFormCalculationExceptionsTest.kt (hunk 2 above)
Comment (grzesiek2010) [reply collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/CatchFormCalculationExceptionsTest.kt]: Yes, done.

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormMediaTest.kt (hunk 3 above)
Comment (grzesiek2010) [reply collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormMediaTest.kt]: Done.

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormAppearancesTest.kt (hunk 5 above)
Comment (grzesiek2010) [reply collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormAppearancesTest.kt]: I spit it into smaller tests you are right the big one didn't make sense.

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormCalculationsTest.kt (hunk 4 above)
Comment (grzesiek2010) [reply collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormCalculationsTest.kt]: > Does Collect actually implement anything to do with random()?

I don't think so. Ok we can gett rid of this.

**Review by seadowg — APPROVED** [review]

**Pull request merged**